exporters/stats/signalfx/src/main/java/io/opencensus/exporter/stats/signalfx/SignalFxSessionAdaptor.java

```diff
@@ -17,172 +17,145 @@
-        data.getAggregationMap().entrySet()) {
-      datapoints.add(
-          DataPoint.newBuilder()
-              .setMetric(view.getName().asString())
-              .setMetricType(metricType)
-              .addAllDimensions(createDimensions(keys, entry.getKey()))
-              .setValue(createDatum(entry.getValue()))
-              .build());
+    DataPoint.Builder shared = DataPoint.newBuilder();
+    shared.setMetric(metricDescriptor.getName());
+    shared.setMetricType(metricType);
+
+    ArrayList<DataPoint> datapoints = Lists.newArrayList();
+    for (TimeSeries timeSeries : metric.getTimeSeriesList()) {
+      DataPoint.Builder builder = shared.clone();
+      builder.addAllDimensions(
+          createDimensions(metricDescriptor.getLabelKeys(), timeSeries.getLabelValues()));
+
+      List<Point> points = timeSeries.getPoints();
+      datapoints.ensureCapacity(datapoints.size() + points.size());
+      for (Point point : points) {
+        datapoints.add(builder.setValue(createDatum(point.getValue())).build());
+      }
     }
     return datapoints;
   }
 
   @VisibleForTesting
   @javax.annotation.Nullable
-  static MetricType getMetricTypeForAggregation(
-      Aggregation aggregation, View.AggregationWindow window) {
-    if (aggregation instanceof Aggregation.Mean || aggregation instanceof Aggregation.LastValue) {
+  static MetricType getType(Type type) {
+    if (type == Type.GAUGE_INT64 || type == Type.GAUGE_DOUBLE) {
       return MetricType.GAUGE;
-    } else if (aggregation instanceof Aggregation.Count || aggregation instanceof Aggregation.Sum) {
-      if (window instanceof View.AggregationWindow.Cumulative) {
-        return MetricType.CUMULATIVE_COUNTER;
-      }
-      // TODO(mpetazzoni): support incremental counters when AggregationWindow.Interval is ready
+    } else if (type == Type.CUMULATIVE_INT64 || type == Type.CUMULATIVE_DOUBLE) {
+      return MetricType.CUMULATIVE_COUNTER;
     }
-
-    // TODO(mpetazzoni): add support for histograms (Aggregation.Distribution).
+    // TODO(mayurkale): decide what to do with Distribution and Summary types.
     return null;
   }
 
   @VisibleForTesting
-  static Iterable<Dimension> createDimensions(
-      List<TagKey> keys, List</*@Nullable*/ TagValue> values) {
-    Preconditions.checkArgument(
-        keys.size() == values.size(), "TagKeys and TagValues don't have the same size.");
+  static Iterable<Dimension> createDimensions(List<LabelKey> keys, List<LabelValue> values) {
     List<Dimension> dimensions = new ArrayList<>(keys.size());
-    for (ListIterator<TagKey> it = keys.listIterator(); it.hasNext(); ) {
-      TagKey key = it.next();
-      TagValue value = values.get(it.previousIndex());
-      if (value == null || Strings.isNullOrEmpty(value.asString())) {
+    for (int i = 0; i < values.size(); i++) {
+      LabelValue value = values.get(i);
+      if (value == null || Strings.isNullOrEmpty(value.getValue())) {
```

Comment:
Apologies I don't recall - is `LabelValue` allowed to be `null`? Or is it just the value of `LabelValue`?